toolkit/tools/imagecustomizerapi/config.go

```diff
@@ -3,15 +3,82 @@
 
 package imagecustomizerapi
 
+import (
+	"fmt"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/sliceutils"
+)
+
 type Config struct {
+	Disks        *[]Disk      `yaml:"Disks"`
 	SystemConfig SystemConfig `yaml:"SystemConfig"`
 }
 
 func (c *Config) IsValid() error {
+	if c.Disks != nil {
+		disks := *c.Disks
+		if len(disks) < 1 {
+			return fmt.Errorf("at least 1 disk must be specified (or the Disks field should be ommited)")
+		}
+		if len(disks) > 1 {
+			return fmt.Errorf("multiple disks is not currently supported")
+		}
+
+		for i, disk := range disks {
+			err := disk.IsValid()
+			if err != nil {
+				return fmt.Errorf("invalid disk at index %d:\n%w", i, err)
+			}
+		}
+	}
+
 	err := c.SystemConfig.IsValid()
 	if err != nil {
 		return err
 	}
 
+	hasDisks := c.Disks != nil
+	hasBootType := c.SystemConfig.BootType != BootTypeUnset
+
+	if hasDisks != hasBootType {
+		return fmt.Errorf("SystemConfig.BootType and Disks must be specified together")
+	}
+
+	// Ensure the correct partitions exist to support the specified the boot type.
+	switch c.SystemConfig.BootType {
+	case BootTypeEfi:
+		hasEsp := sliceutils.ContainsFunc(*c.Disks, func(disk Disk) bool {
```

Comment:
just curious, is the use of ContainsFunction/Value here equivalent to nexted loops? Is there an advantage of using lambdas here? (in C++ this would typically be a bit more expensive than just doing 3 nested loops)

Reply:
Yes, this could also be implemented with nested loops. It is just a matter of style preference.

(This code os only run once per program execution and works on very short lists. So, it doesn't need to be optimized.)